src/App.jsx

```diff
@@ -1,17 +1,44 @@
+import React, { useState } from 'react';
 import './App.css';
+import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
+import chatMessages from './data/messages.json';
 
 const App = () => {
+  // const [entries, setEntries] = useState(chatMessages);
+  const [entries, setEntries] = useState(
+  chatMessages.map((msg) => ({ ...msg, liked: msg.liked ?? false }))
+);
+
+  const toggleLike = (id) => {
+    const updatedEntries = entries.map((entry) => {
+      if (entry.id === id) {
+        return { ...entry, liked: !entry.liked };
+      }
+      return entry;
+    });
+    setEntries(updatedEntries);
+  };
+
+  const totalLikes = entries.filter((entry) => entry.liked).length;
```

Comment:
Great idea to filter down the data to the liked messages and use the list length as the count!

This approach works, but you'll typically see JS devs use `reduce` to calculate a value like this:

```js
  const calculateTotalLikeCount = (chatData) => {
    return entries.reduce((acc, entry) => {
      return entry.liked ? acc + 1 : acc;
    }, 0);
  };
```